kernel/Kconfig

```diff
@@ -576,6 +576,25 @@ config STACK_CANARIES
 	option has no effect.
 endmenu
 
+menu "Memory Domain APIs"
+
+config MEMORY_DOMAIN
+	bool
+	prompt "Memory domain APIs support"
+	depends on HAS_SYSMPU || ARM_MPU
```

Comment:
I don't think these arch-specific "depends on" should be here. Remove. See below, I'd like references to the ARM code removed from kernel/ and an interface declared instead.

Reply:
You could depend on CONFIG_USERSPACE though

Reply:
Or better yet, just make memory domains part of CONFIG_USERSPACE. I don't think we have a strong need for it to be its own config.

Reply:
I've removed the dependency of HAS_SYSMPU and ARM_MPU.
But I decide to keep CONFIG_MEMORY_DOMAIN for now, since the CONFIG_USERSPACE can't work on arch arm yet.